x/poolmanager/router.go

```diff
@@ -305,6 +369,65 @@ func (k Keeper) RouteExactAmountOut(ctx sdk.Context,
 	return tokenInAmount, nil
 }
 
+// SplitRouteExactAmountOut routes the swap across multiple multihop paths
+// to get the desired token in. This is useful for achieving the most optimal execution. However, note that the responsibility
+// of determining the optimal split is left to the client. This method simply routes the swap across the given routes.
+//
+// It performs the price impact protection check on the combination of tokens in from all multihop paths. The given tokenInMaxAmount
+// is used for comparison.
```

Comment:
nit: consider adding more docs (doesn't have to be as recommended), took some time for me to understand why this is needed or how this is different from `RouteExactAmountOut` 
```suggestion
// SplitRouteExactAmountOut routes the swap across multiple multihop paths
// to get the desired token in, where each route would be multi-hop routes. This allows batching multihop swaps.
//
// This is useful for achieving the most optimal execution. However, note that the responsibility
// of determining the optimal split is left to the client. This method simply routes the swap across the given routes.
//
// It performs the price impact protection check on the combination of tokens in from all multihop paths. The given tokenInMaxAmount
// is used for comparison.
```

Reply:
Expanded! However, I'm not sure if calling this batching is correct because we only allow routes on the that start with the same token in and end with the same token out